pam/integration-tests/ssh_test.go

```diff
@@ -0,0 +1,557 @@
+			user:         "user-needs-reset2",
+			tapeSettings: []tapeSetting{{vhsHeight, 900}},
+		},
+
+		"Prevent user from switching username": {
+			tape:         "switch_preset_username",
+			tapeSettings: []tapeSetting{{vhsHeight, 950}},
+		},
+
+		"Exit authd if local broker is selected": {
+			tape:            "local_broker",
+			ignoreUserCheck: true,
+		},
+		"Exit if user is not pre-checked on ssh service": {
+			tape:            "local_ssh",
+			user:            "user-integration-ssh-service",
+			pamServiceName:  "sshd",
+			ignoreUserCheck: true,
+		},
+		"Exit authd if user sigints": {
+			tape:            "sigint",
+			ignoreUserCheck: true,
+		},
+	}
+	for name, tc := range tests {
+		if defaultSSHDPort != "" {
+			name = fmt.Sprintf("%s on shared SSHd", name)
+		}
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			socketPath := defaultSocketPath
+			gpasswdOutput := defaultGPasswdOutput
+			if tc.wantLocalGroups {
+				var groupsFile string
+				gpasswdOutput, groupsFile = prepareGPasswdFiles(t)
+				socketPath = runAuthd(t, gpasswdOutput, groupsFile, true)
+			}
+
+			user := tc.user
+			if user == "" {
+				user = "user-integration-pre-check-" + strings.ReplaceAll(
+					strings.ToLower(filepath.Base(t.Name())), "_", "-")
+			}
+
+			sshdPort := defaultSSHDPort
+			userHome := defaultUserHome
+			if defaultSSHDPort == "" || tc.wantLocalGroups || tc.interactiveShell {
+				serviceFile := createSshdServiceFile(t, execModule, execChild, socketPath)
+				sshdPort, userHome = startSSHdForTest(t, serviceFile, sshdHostKey, user,
+					sshdPreloadLibrary, tc.daemonizeSSHd, tc.interactiveShell)
+			}
+
+			knownHost := filepath.Join(t.TempDir(), "known_hosts")
+			err := os.WriteFile(knownHost, []byte(
+				fmt.Sprintf("[localhost]:%s %s", sshdPort, pubKey),
+			), 0600)
+			require.NoError(t, err, "Setup: can't create known hosts file")
+
+			outDir := t.TempDir()
+			td := newTapeData(tc.tape, append(defaultTapeSettings, tc.tapeSettings...)...)
+			td.Command = tapeCommand
+			td.CommandSleep = defaultSleepValues[authdSleepLong]
+			td.Env[pam_test.RunnerEnvSupportsConversation] = "1"
+			td.Env["HOME"] = userHome
+			td.Env["AUTHD_PAM_SSH_USER"] = user
+			td.Env["AUTHD_PAM_SSH_ARGS"] = strings.Join([]string{
+				"-p", sshdPort,
+				"-F", os.DevNull,
+				"-i", os.DevNull,
+				"-o", "ServerAliveInterval=300",
+				"-o", "PasswordAuthentication=no",
+				"-o", "PubkeyAuthentication=no",
+				"-o", "UserKnownHostsFile=" + knownHost,
+			}, " ")
+			td.RunVhs(t, "ssh", outDir, nil)
+			got := sanitizeGoldenFile(t, td, outDir)
+			want := testutils.LoadWithUpdateFromGolden(t, got)
+
+			require.Equal(t, want, got, "Output of tape %q does not match golden file", tc.tape)
+			userEnv := fmt.Sprintf("USER=%s", user)
+			if !tc.ignoreUserCheck {
+				require.Contains(t, got, userEnv, "Logged in user does not matches")
+			} else {
+				require.NotContains(t, got, userEnv, "Should not have a logged in user")
+			}
```

Comment:
Yeah, so... Initially I wasn't doing the check then i thought it's just better to cover both paths, so can be renamed to `wantLoggedInUser` indeed (using `want` as I learned that is this team prefix for expectations :))

Reply:
> Initially I wasn't doing the check then i thought it's just better to cover both paths, so can be renamed to wantLoggedInUser indeed

:+1: 

> (using want as I learned that is this team prefix for expectations :))

That's something I wanted to address in the team: I think "want" is a bit ambiguous, because it can be understood as "the test case wants this precondition" or as "the test case wants this outcome". I think we do in fact use it with the former meaning sometimes, for example in this same test you define `wantLocalGroups`, which, when set, [makes the necessary preparations](https://github.com/ubuntu/authd/blob/main/pam/integration-tests/ssh_test.go#L208-L212) to support adding the user to local groups without interfering with other test cases (if I understand correctly).

Reply:
Yeah, I kinda agree, in my initial PR's I was always using `expectsFoo` but I was asked to change it, so... I feel we won't change that as per consistency we prefer to reuse what's used in all the projects